lib/route.js

```diff
@@ -504,6 +516,19 @@ function removeTrackedSocket () {
   this.keepAliveConnections.delete(this.socket)
 }
 
+function isHttpVersionValid (req, context) {
+  const { http2, https } = context.server.initialConfig
+  if (http2 === true && https && https.allowHTTP1 === true) {
+    return ['1.0', '1.1', '2.0'].includes(req.httpVersion)
```

Comment:
maybe an old `req.httpVersion === '1.0' || req.httpVersion === '1.1' ...` solves the perf issue?

Reply:
I've tried a couple of variation like that but it did not improve